Qt35.py

```diff
@@ -0,0 +1,318 @@
+
+        grid.addWidget(self.pushButton_4, 4, 1)
+
+        self.progressBar = QtWidgets.QProgressBar(self)
+        self.progressBar.setGeometry(QtCore.QRect(10, 640, 841, 20))
+        self.progressBar.setProperty("value", 24)
+        self.progressBar.setObjectName("progressBar")
+
+        grid.addWidget(self.progressBar, 1, 10)
+
+        self.dateTimeEdit = QtWidgets.QDateTimeEdit(self)
+        self.dateTimeEdit.setGeometry(QtCore.QRect(890, 20, 194, 22))
+        self.dateTimeEdit.setObjectName("dateTimeEdit")
+
+        grid.addWidget(self.dateTimeEdit, 1, 10)
+
+        self.dateTimeEdit_2 = QtWidgets.QDateTimeEdit(self)
+        self.dateTimeEdit_2.setGeometry(QtCore.QRect(890, 50, 194, 22))
+        self.dateTimeEdit_2.setObjectName("dateTimeEdit_2")
+
+        grid.addWidget(self.dateTimeEdit_2, 2, 10)
+
+        self.win_text = QtWidgets.QTextBrowser(self)
+        self.win_text.setGeometry(QtCore.QRect(100, 100, 1940, 2200))
+
+        grid.addWidget(self.dateTimeEdit_2, 3, 3)
+
+        self.label = QLabel(self)
+
+        grid.addWidget(self.label, 4, 4)
+
+        self.setWindowTitle("日")
+
+        self.setWindowTitle("Widget")
+        self.tideButton.setText("潮位")
+        self.pushButton_2.setText("波浪")
+        self.pushButton_3.setText("风")
+        self.pushButton_4.setText("潮流")
+
+        self.show()
+
+    def change_switch(self, state):
+        if state == self.is_plot:
+            self.plot()
+        else:
+            pass
+
+    def show_it(self):
+        self.switch_plot = QtWidgets.QCheckBox('是否画对应潮位图', self)
+        self.switch_plot.setGeometry(QtCore.QRect(700, 30, 150, 40))
+
+        self.switch_plot.stateChanged.connect(self.change_switch)
+        self.switch_plot.toggle()
+
+    def plot(self):
+        self.pixmap = QtGui.QPixmap('x.png')
+        self.label.setPixmap(self.pixmap)
+
+    def get_time(self):
+        self.begin = self.dateTimeEdit.dateTime()
+        self.end = self.dateTimeEdit_2.dateTime()
+
+    def open_init_file(self):
+        self.show_it()
+        self.fileName = QFileDialog.getOpenFileName(self, "选取文件", "E:",
+                                                    "Excel文件(*.xlsx);;97-03表格文档 (*.xls)")  # 设置文件扩展名过滤,注意用双分号间隔
+
+        if self.is_plot:
+            self.plot()
+
+
+class Tide(object):
+    def __init__(self, filename, time1=datetime.datetime(2000, 1, 1, 00, 0),
+                 time2=datetime.datetime(2016, 10, 16, 5, 0)):
+        self.data = {}
+        self.year = {}
+        self.month = {}
+        self.day = {}
+        for s in xlrd.open_workbook(filename=filename).sheet_names():
+            try:
+                self.tide_sheet = pandas.read_excel(filename, sheetname=s)
+                if 'tide' in self.tide_sheet.columns.values:
+                    if 'time' in self.tide_sheet.columns.values:
+                        pass
+                    else:
+                        say_out('Please Check Your Format of the Sheet')
+            except:
```

Comment:
- To catch any exception use the `except Exception`, instead of `except BaseException` or just `except`.